core/include/drivers/firewall_device.h

```diff
@@ -0,0 +1,224 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2024, STMicroelectronics
+ */
+
+#ifndef __DRIVERS_FIREWALL_DEVICE_H
+#define __DRIVERS_FIREWALL_DEVICE_H
+
+#include <compiler.h>
+#include <stdint.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+#include <util.h>
+
+/* Opaque reference to firewall_controller */
+struct firewall_controller;
+
+/**
+ * struct firewall_query - Information on a device's firewall. Each device can
+ * have more than one firewall.
```

Comment:
I don't think the sentence `Each device can have more than one firewall` is useful when describing this function.

Reply:
I agree